Mappers/HttpMapper.php

```diff
@@ -171,14 +204,18 @@ protected function getHttpClient()
     }
 
     /**
-     * @param HttpClient $httpClient
-     * @return HttpMapper
+     * @param string $content
+     * @param $typeContent
+     * @return array
+     * @throws InfrastructureException
      */
-    private function setHttpClient($httpClient)
+    private function parse(string $content, $typeContent)
     {
-        $this->httpClient = $httpClient;
+        if ($typeContent == self::CONTENT_TYPE_JSON) {
+            return json_decode($content, true);
+        }
 
-        return $this;
+        throw new InfrastructureException("Illegal Content-Type: " . $typeContent);
```

Comment:
Why illegal? we just don't support it. We use third party, and this is weird to say that they send illegal content type.

Reply:
$contentType can be override